compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1033,6 +1033,25 @@ impl<'context> Elaborator<'context> {
             return;
         };
 
+        if the_trait.trait_bounds.is_empty() {
+            return;
+        }
+
+        let impl_trait = the_trait.name.to_string();
+        let the_trait_file = the_trait.location.file;
+
+        let mut bindings = TypeBindings::new();
+        for (param, arg) in the_trait.generics.iter().zip(trait_impl.resolved_trait_generics.iter())
+        {
+            // Avoid binding t = t
+            if !arg.occurs(param.type_var.id()) {
+                bindings.insert(
+                    param.type_var.id(),
+                    (param.type_var.clone(), param.kind(), arg.clone()),
+                );
+            }
+        }
```

Comment:
I see. I'll try to think of some code that would fail without adding the associated types (without it I can't be sure I implemented it correctly).

Reply:
I pushed a bunch of refactors so the "same length assertion" will happen now.

To check associated types now we'd need to call `bind_named_generics` but I don't know where to get `args: &[NamedType]` from.

Reply:
There's `the_trait.associated_types` for the generics and `trait_impl.associated_types` for the bindings

Reply:
Also - instead of writing these special methods, I wonder if we can make use of `resolve_type_args` which should accept any generic thing with some `GenericTypeArgs`. One such thing for the `item: impl Generic` parameter is a TraitId.

Reply:
> trait_impl.associated_types

It seems those are UnresovledTypes...

I tried to find some code that doesn't work with associated types and couldn't. I think it's because if you have something like this:

```noir
impl Bar<Field, C = i32> for () {}
```

The associated types specified in `Bar` are checked against `Bar`, but they don't show up in parent traits. So for checking a parent trait we don't need to analyze the associated types. But not sure!

I'll try to use `resolve_type_args`.

Reply:
I tested this and couldn't break it either, I'll just resolve this comment 👍